yamlparse-applicative/src/YamlParse/Applicative/Pretty.hs

```diff
@@ -58,32 +77,32 @@ comment :: Text -> Comments
 comment t = Comments $ map pretty $ T.lines t
 
 -- | Prettyprint a 'Schema'
-schemaDoc :: Schema -> Doc ()
+schemaDoc :: Schema -> Doc Color
 schemaDoc = go emptyComments
   where
-    go :: Comments -> Schema -> Doc ()
+    go :: Comments -> Schema -> Doc Color
     go cs =
       let g = go cs
           ge = go emptyComments
-          mkComment :: Doc () -> Doc ()
+          mkComment :: Doc Color -> Doc Color
           mkComment = ("# " <>)
-          mkCommentsMDoc :: Comments -> Maybe (Doc ())
+          mkCommentsMDoc :: Comments -> Maybe (Doc Color)
           mkCommentsMDoc = \case
             Comments [] -> Nothing
-            Comments l -> Just $ align $ vsep $ map mkComment l
+            Comments l -> Just $ align $ vsep $ map ((annotate Gray) . mkComment) l
           addMComment :: Comments -> Maybe Text -> Comments
           addMComment c = \case
             Nothing -> c
             Just t -> c <> comment t
-          e :: Doc () -> Comments -> Doc ()
+          e :: Doc Color -> Comments -> Doc Color
           e s cs' =
             case mkCommentsMDoc cs' of
-              Nothing -> s
-              Just cd -> vsep [cd, s]
+              Nothing -> annotate Yellow s
+              Just cd -> vsep [cd, annotate Yellow s]
        in \case
-            EmptySchema -> e "# Nothing to parse" cs
+            EmptySchema -> e emptyDoc $ addMComment cs $ Just "Nothing to parse"
             AnySchema -> e "<any>" cs
-            ExactSchema t -> e (pretty t) cs
+            ExactSchema t -> e (pretty t <+> annotate Gray "(exact)") cs
```

Comment:
I like this change, but :

1. The colours don't work it seems:
![colour_000](https://user-images.githubusercontent.com/3521180/82753780-a1b16900-9dc8-11ea-9ed9-3f12ac97b66c.png)
2. Could we put that in a comment instead of in parentheses:
```
e (pretty t) (cs <> comment "exact")
```

Reply:
Hm, for me it dims `(exact)` (blend yellow with gray) which i found to be looking nicely.

Even without colors, I find the distinction pretty clear so don't see an issue for colorblind people. But can change it for stylistic reasons if you like

Reply:
![comment](https://user-images.githubusercontent.com/3521180/82753921-e4c00c00-9dc9-11ea-8a4a-68cab1b16e6d.png)
How do you like this?

```
            ExactSchema t ->
              let mExactCommentDoc = mkCommentsMDoc $ comment "(exact)"
               in ( case mExactCommentDoc of
                      Nothing -> id
                      Just cd -> (<+> cd)
                  )
                    $ e (pretty t) cs
```

Reply:
Done using `fromJust`